### PR TITLE
[improvement](statistics)Support get index row count and table delta rows.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Provide get row count for a given index id and get table delta rows API in TableStatsMeta class. This is for Nereids stats calculator to fetch row count.

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterColumnStatsStmt.java [hunk 1]
```diff
@@ -134,6 +134,10 @@ public void analyze(Analyzer analyzer) throws UserException {
             throw new AnalysisException(optional.get() + " is invalid statistics");
         }
 
+        if (!properties.containsKey(StatsType.ROW_COUNT.getValue())) {
+            throw new AnalysisException("Must set row_count");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterColumnStatsStmt.java:138]: add a more formal error msg? hint how to add row count

**Review by englefly — DISMISSED** [review]

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run cloud_p0

**Comment by Jibing-Li** [comment]
run cloud_p0

**Review by morrySnow — APPROVED** [review]

**Review by zfr9527 — APPROVED** [review]
LGTM

**Pull request merged**